packages/playground/assets/index.html

```diff
@@ -276,6 +284,16 @@ <h3 class="raw-query"></h3>
   text('.import-meta-url', metaUrl)
   document.querySelector('.import-meta-url-img').src = metaUrl
 
+  const metaUrlDep = new URL('@/asset.png', import.meta.url)
```

Comment:
What do you think about the https://github.com/vitejs/vite/pull/7837#issuecomment-1104672574. I think is better to remove `import.meta.url` in absolute path.

Maybe you can use this regexp

```js
const assetImportMetaUrlRE =
  /\bnew\s+URL\s*\(\s*('[^']+'|"[^"]+"|`[^`]+`)\s*(,|,\s*import\.meta\.url){0,1}\s*\)/g
```

https://regex101.com/r/mSdtWD/1

Reply:
Keeping `import.meta.url` for resolving package assets is important since it gives Vite parity with webpack in this scenario.

When `import.meta.url` isn't included, I'm wondering if a custom protocol could be used instead. For example `new URL('vite:@/asset.png'). I kind of like this because this allows the asset to use a valid absolute URL and is similar to how Node defines its `node:` URLs for its builtins.

Reply:
For now I think we can also have `import.meta.url` as is. It wouldn't be used when resolving absolute or aliased path, but perhaps we can have another discussion to allow without `import.meta.url` too.